apps/aecore/src/aec_metrics_rpt_dest.erl

```diff
@@ -0,0 +1,185 @@
+    end.
+
+init(Tab) ->
+    {ok, #st{tab = Tab}}.
+
+handle_call({add_destination, Name, DP, Dest}, _From, #st{tab = T} = St) ->
+    case ets:lookup(T, {Name, DP}) of
+        [#metric{destinations = Ds} = M] ->
+            case lists:member(Dest, Ds) of
+                true ->
+                    {reply, ok, St};
+                false ->
+                    M1 = M#metric{destinations = [Dest|Ds]},
+                    ets:insert(T, M1),
+                    {reply, ok, St}
+            end;
+        [] ->
+            ets:insert(T, #metric{key = {Name, DP},
+                                  destinations = [Dest]}),
+            {reply, ok, St}
+    end;
+handle_call({del_destination, Name, DP, Dest}, _From, #st{tab = T} = St) ->
+    Res = case ets:lookup(T, {Name, DP}) of
+              [#metric{destinations = Ds} = M] ->
+                  case lists:member(Dest, Ds) of
+                      true ->
+                          ets:insert(T, M#metric{destinations = Ds -- [Dest]}),
+                          true;
+                false ->
+                    false
+                  end;
+              [] ->
+                  false
+          end,
+    {reply, Res, St};
+handle_call(check_config, _From, #st{tab = T} = St) ->
+    populate_tab(T),
+    lager:debug("Dest tab = ~p", [ets:tab2list(T)]),
+    {reply, ok, St};
+handle_call(_Req, _From, St) ->
+    {reply, {error, unknown_call}, St}.
+
+handle_cast(_Msg, St) ->
+    {noreply, St}.
+
+handle_info(_Msg, St) ->
+    {noreply, St}.
+
+terminate(_Reason, _St) ->
+    ok.
+
+code_change(_From, St, _Extra) ->
+    {ok, St}.
+
+
+%% ======================================================================
+%% Private functions
+%% ======================================================================
+
+ensure_tab() ->
+    case ets:info(?TAB, size) of
+        undefined ->
+            T = ets:new(?TAB, [set, public, named_table,
+                               {keypos, #metric.key},
+                               {heir, self(), []}]),
+            T;
+        _ ->
+            ?TAB
+    end.
+
+populate_tab(Tab) ->
+    case aeu_env:user_map([<<"metrics">>,<<"rules">>]) of
+        {ok, Rules} ->
+            lists:foreach(
+              fun(#{} = R) ->
+                      Name = maps:get(<<"name">>, R, <<"*">>),
+                      Type = maps:get(<<"type">>, R, <<"*">>),
+                      DPs  = maps:get(<<"datapoints">>, R, <<"default">>),
+                      Actions = maps:get(<<"actions">>, R, <<"log,send">>),
+                      Cmd = <<"type=",Type/binary,"/",
+                              Name/binary,"/",
+                              DPs/binary>>,
+                      try
+                          Found = aec_metrics_lib:find_entries_cmd([Cmd]),
+                          make_destinations(Found, Actions, Tab)
+                      catch
+                          error:E ->
+                              lager:error("CRASH: ~p~n~p~n",
```

Comment:
Can we somehow remove the newlines here?

see https://github.com/aeternity/epoch/pull/456